app/policies/photo_policy.rb

```diff
@@ -0,0 +1,42 @@
+# app/policies/photo_policy.rb
+
+class PhotoPolicy < ApplicationPolicy
+  attr_reader :user, :photo
+
+  def initialize(user, photo)
+    @user = user
+    @photo = photo
+  end
+
+  def show?
+    user == photo.owner || !photo.owner.private? || photo.owner.followers.include?(user)
+  end
+
+  def create?
+    true
+  end
+
+  def new?
+    create?
+  end
+
+  def update?
+    user == photo.owner
+  end
+
+  def edit?
+    update?
+  end
+
+  def destroy?
+    user == photo.owner
+  end
+
+  class Scope < Scope
+    def resolve
+      scope.all.select do |photo|
```

Comment:
fwiw using a select do block here wont scale or work with pagination since we have to load all the objects into memory.